axelor-human-resource/src/main/java/com/axelor/apps/hr/service/project/ProjectTaskHRServiceImpl.java

```diff
@@ -0,0 +1,224 @@
+import com.axelor.apps.project.db.ProjectTask;
+import com.axelor.apps.project.db.Sprint;
+import com.axelor.apps.project.db.SprintPeriod;
+import com.axelor.apps.project.db.repo.ProjectRepository;
+import com.axelor.apps.project.db.repo.ProjectTaskRepository;
+import com.axelor.apps.project.db.repo.TaskStatusProgressByCategoryRepository;
+import com.axelor.apps.project.service.ProjectTaskServiceImpl;
+import com.axelor.apps.project.service.TaskStatusToolService;
+import com.axelor.apps.project.service.app.AppProjectService;
+import com.axelor.auth.db.User;
+import com.google.inject.Inject;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.collections.CollectionUtils;
+
+public class ProjectTaskHRServiceImpl extends ProjectTaskServiceImpl
+    implements ProjectTaskHRService {
+
+  @Inject
+  public ProjectTaskHRServiceImpl(
+      ProjectTaskRepository projectTaskRepo,
+      FrequencyRepository frequencyRepo,
+      FrequencyService frequencyService,
+      AppBaseService appBaseService,
+      ProjectRepository projectRepository,
+      AppProjectService appProjectService,
+      TaskStatusToolService taskStatusToolService,
+      TaskStatusProgressByCategoryRepository taskStatusProgressByCategoryRepository) {
+
+    super(
+        projectTaskRepo,
+        frequencyRepo,
+        frequencyService,
+        appBaseService,
+        projectRepository,
+        appProjectService,
+        taskStatusToolService,
+        taskStatusProgressByCategoryRepository);
+  }
+
+  @Override
+  public List<ProjectPlanningTime> updateProjectPlanningTime(
+      ProjectTask projectTask, ProjectTask projectTaskDb) {
+
+    List<ProjectPlanningTime> projectPlanningTimeList = projectTask.getProjectPlanningTimeList();
+
+    if (CollectionUtils.isEmpty(projectPlanningTimeList)) {
+      ProjectPlanningTime projectPlanningTime =
+          createProjectPlanningTime(projectTask, projectTask.getSprint());
+
+      if (projectPlanningTime != null) {
+        projectPlanningTimeList.add(projectPlanningTime);
+      }
+
+      return projectPlanningTimeList;
+    }
+
+    if (isInvalidInput(projectTask, projectTaskDb)) {
+      return projectPlanningTimeList;
+    }
+
+    Employee employee =
+        Optional.ofNullable(projectTask)
+            .map(ProjectTask::getAssignedTo)
+            .map(User::getEmployee)
+            .orElse(null);
+    Employee dbEmployee =
+        Optional.ofNullable(projectTaskDb)
+            .map(ProjectTask::getAssignedTo)
+            .map(User::getEmployee)
+            .orElse(null);
+    SprintPeriod sprintPeriod = projectTask.getSprint().getSprintPeriod();
+    SprintPeriod dbSprintPeriod = projectTaskDb.getSprint().getSprintPeriod();
+
+    LocalDateTime fromDateTime = sprintPeriod.getFromDate().atStartOfDay();
+    LocalDateTime toDateTime = sprintPeriod.getToDate().atStartOfDay();
```

Comment:
Use Optional on these 2 please

Reply:
fromDate and toDate are required fields in sprintPeriod